src/pydata_sphinx_theme/theme/pydata_sphinx_theme/search.html

```diff
@@ -9,6 +9,7 @@ <h1>{{ _("Search") }}</h1>
         <p>{% trans %}Please activate JavaScript to enable the search functionality.{% endtrans %}</p>
       </div>
     </noscript>
+    {%- set search_input_id="search-input-results-page" -%}
```

Comment:
The search input can appear twice in the DOM (once in the hidden modal dialog) and once at the top of the search results page. So it's necessary to disambiguate them if using a label element.

Hmm, but I just realized that this doesn't work for the persistent search bar.

TODO: fix this.